website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 Tecnativa - Alexandre Díaz
+// License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+odoo.define("website_snippet_carousel_product.s_product_carousel", function (require) {
+    "use strict";
+
+    var core = require("web.core");
+    var sAnimation = require("website.content.snippets.animation");
+
+    var _t = core._t;
+
+    sAnimation.registry.js_product_carousel = sAnimation.Class.extend({
+        selector: ".js_product_carousel",
+
+        _refreshAnimations: function () {
+            this.trigger_up("animation_start_demand", {
+                editableMode: false,
+                $target: this.$target.find(".product_carousel_slider"),
+            });
+        },
+
+        /**
+         * @override
+         */
+        start: function () {
+            var self = this;
+            var limit = Number(this.$target.attr("data-products-limit")) || 12;
+            var domain = this.$target.attr("data-domain") || false;
```

Comment:
```suggestion
            var domain = this.$target.attr("data-domain") || "[]";
```

Otherwise the `JSON.parse()` call you do later will fail.